tests/test_validators.py

```diff
@@ -31,6 +32,50 @@ def test_invalid_url(self):
             self.form.process(url=url)
             self.assertEqual(self.form.validate(), False)
 
+    def test_public_email_domain(self):
+        with self.app.test_request_context('/'):
+            # both valid
+            self.webmail_form.process(
+                webmail_domain='gmail.com',
+                not_webmail_domain='foobar.com'
+            )
+            self.assertTrue(self.webmail_form.validate())
+
+            # Emoji domain, should not raise any exception
+            self.webmail_form.process(
+                webmail_domain='gmail.com',
+                not_webmail_domain='i❤.ws'
+            )
+            self.assertTrue(self.webmail_form.validate())
+
+            # both invalid
+            self.webmail_form.process(
+                webmail_domain='foobar.com',
+                not_webmail_domain='yahoo.com'
+            )
+            self.assertFalse(self.webmail_form.validate())
+            self.assertIn('webmail_domain', self.webmail_form.errors)
+            self.assertIn('not_webmail_domain', self.webmail_form.errors)
+
+            # one valid, one invalid
+            self.webmail_form.process(
+                webmail_domain='gmail.com',
+                not_webmail_domain='yahoo.com'
+            )
+            self.assertFalse(self.webmail_form.validate())
+            self.assertNotIn('webmail_domain', self.webmail_form.errors)
+            self.assertIn('not_webmail_domain', self.webmail_form.errors)
+
+            # these domain names don't exist, MX lookup will fail
+            # So, webmail_domain should fail, but not_webmail_domain should pass
+            self.webmail_form.process(
+                webmail_domain='9vGw1TMChQWH.com',
+                not_webmail_domain='9vGw1TMChQWH.com'
+            )
+            self.assertFalse(self.webmail_form.validate())
+            self.assertIn('webmail_domain', self.webmail_form.errors)
+            self.assertNotIn('not_webmail_domain', self.webmail_form.errors)
+
```

Comment:
Please pass the same values to both validators in all tests, to confirm that they always exhibit opposite responses.

Reply:
Please test the MXLookupException handler as well. Accept a timeout parameter in both validators, and in one test change the timeout to a very small number.